### ISSUE TITLE
Question about training code

### ISSUE THREAD

**Issue by dgjung0220** [xuan611/sRGB-Real-Noise-Synthesizing#1]
Hello. Thank you for providing the code. I have a question about your code, specifically in main_train_jointly.py. Why is camera_name being used? Do we need to train separately for each camera name? I thought we could train the noise altogether since we have the Gain Estimation network. Also, there seems to be an issue with the parameters of the lr_scheduler's MultiStepLR. The milestones of the MultiStep LR should be a list, but the current code has a single integer value, which would cause the training to not work. Could you please provide the actual milestones values used in the paper's experiments?

**Comment by xuan611** [comment]
Hi, thanks for your questions. It seems you have got two problems: (1) Do we need to train separately for each camera and (2) the issues in the code. For the first question, we do need to train separately for each camera even though we have the Gain Estimation network.

**Comment by zfu006** [comment]
Hi, thanks for your questions. It seems you have got two problems: (1) Do we need to train separately for each camera and (2) the issues in the code. For the first question, we do need to train separately for each camera even though we have the Gain Estimation network. The reason for that designed training scheme is that the noise level functions for different cameras are not the same. For example, in the sRGB domain, the noise level functions of IPhone and Google Pixel are different. You can find the results in papers like Modeling sRGB Camera Noise with Normalizing Flows, CVPR 2022. So the main issue is that for the Noise level Prection network if we feed a gain factor (estimated from the Gain Estimation network or set manually) and a clean image into the Noise level Prection network, the output noise level maps will be the same for different cameras. However, these two noise level maps are supposed to be different. That is why we do not train cameras altogether. For the second problem, I will update the code as soon as possible.

**Comment by dgjung0220** [comment]
Thank you for your answer, so in this paper that you mentioned about denoising performance Table2, do you use few paired images with only one camera model? If so, the denoising performance is so promissing and it would be interesting work! I’ll hope to update your code soon, thanks.

**Comment by zfu006** [comment]
Yes, in table 2, for NeCA-S, we train the model using few paired images from N6. I've fixed the code, you may run main_train_jointly.py.

**Comment by dgjung0220** [comment]
Thank you for your response and code update. I have one last question. What about NeCA-W? Was it trained for each camera model, and its performance measured on the SIDD benchmark with a single camera model?

**Comment by zfu006** [comment]
Yes, NeCA-W is trained for each camera model. However, when evaluating NeCA-W, we use multiple models. For an image from the SIDD training set, for example, an image captured by IPhone, we use the model trained on IPhone and synthesize the noise. So the performance of NeCA-W is measured with five camera models. This is a good question that the reviewers also asked us, they said this experiment settings may be unfair since we use multiple models. So we think that can be further work to combine multiple models into one model.

**Comment by dgjung0220** [comment]
Thank you for answering my questions. I understood your research well. I close this issue.
